src/main/java/hudson/plugins/git/BranchSpec.java

```diff
@@ -78,6 +81,27 @@ public boolean matches(String ref, EnvVars env) {
     }
 
     /**
+     * Compare the configured pattern to a git branch defined by the a repository name and the branch name itself.
+     */
+    public boolean matchesRepositoryBranch(String repositoryName, String branchName) {
```

Comment:
Isn't there already a branch name expansion function elsewhere in the plugin that expands environment variables in the branch name, decides if the branch name is a regex, performs matching, converts wildcards, and more?  Could we use that instead (so that the knowledge of the "leading colon starts a regex" is limited to one location (or to fewer locations)?